fcrepo/fcrepo
# FCREPO-3409 - DC membership index

**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3409

# What does this Pull Request do?
Adds and populates a membership index, which tracks the membership history of resources.

I'm not sure if a message should be emitted when membership changes, but there is not an existing test indicating so.

# How should this be tested?

The application's behaviors should be unchanged from before since the service to add membership to responses has not been completed. I could update this PR to add it, but it was a separate ticket https://jira.lyrasis.org/browse/FCREPO-3165

So it should build and the new tests should pass.

# Interested parties
@fcrepo4/committers
